lib/iris/analysis/cartography.py

```diff
@@ -70,14 +70,37 @@ def wrap_lons(lons, base, period):
 
 def unrotate_pole(rotated_lons, rotated_lats, pole_lon, pole_lat):
     """
-    Convert rotated-pole lons and lats to unrotated ones.
+    Convert arrays of rotated-pole longitudes and latitudes to unrotated
+    arrays of longitudes and latitudes. The values of ``pole_lon`` and
+    ``pole_lat`` should describe the location of the rotated pole that
+    describes the arrays of rotated-pole longitudes and latitudes.
+
+    As the arrays of rotated-pole longitudes and latitudes must describe a
+    rectilinear grid, the arrays of rotated-pole longitudes and latitudes must
+    be of the same length as each other.
 
     Example::
 
         lons, lats = unrotate_pole(grid_lons, grid_lats, pole_lon, pole_lat)
```

Comment:
I wonder whether it's also worth changing the words "grid_lats" and "grid_lons". It implies that a grid of coordinates is being (un)rotated but this isn't necessarily the case.

I'm just concerned that the following difference is clear:
```
>>> x = np.array([1, 2, 3, 4]).reshape(2, 2)
>>> y = np.array([0, 5, 10, 15]).reshape(2, 2)
>>> lons, lats = iris.analysis.cartography.rotate_pole(x, y, 20, 40)
>>> print lons
[[ 151.8229954   150.41532154]
 [ 148.63426722  146.2567781 ]]
```

and
```
>>> x_grid, y_grid = np.meshgrid(x, y)
>>> lons_grid, lats_grid = iris.analysis.cartography.rotate_pole(x_grid, y_grid, 20, 40)
>>> print lons_grid
[[ 151.8229954   153.18407264  154.56272063  155.95851087]
 [ 148.94908795  150.41532154  151.90509014  153.41798484]
 [ 145.44282062  147.02245908  148.63426722  150.27799649]
 [ 141.03227915  142.72882067  144.47023039  146.2567781 ]]
```